datamol/io.py

```diff
@@ -544,3 +545,129 @@ def to_xlsx(
 
     with fsspec.open(urlpath, mode="wb") as f:
         PandasTools.SaveXlsxFromFrame(mols, f, molCol=mol_column, size=mol_size)
+
+
+EXTENSIONS_DICT = {
+    "csv": [
+        ".csv",
+        ".csv.gz",
+        ".csv.bz2",
+        ".csv.zip",
+        ".csv.xz",
+        ".csv.zst",
+        ".csv.tar",
+        ".csv.tar.gz",
+        ".csv.tar.xz",
+        ".csv.tar.bz2",
+    ],
+    "excel": [".xlsx"],
+    "parquet": [".parquet"],
+    "json": [
+        ".json",
+        ".json.gz",
+        ".json.bz2",
+        ".json.zip",
+        ".json.xz",
+        ".json.zst",
+        ".json.tar",
+        ".json.tar.gz",
+        ".json.tar.xz",
+        ".json.tar.bz2",
+    ],
+    "sdf": [
+        ".sdf",
+        ".sdf.gz",
+    ],
+}
+
+
+def _guess_filetype(path: str):
+    """Return a filetype given an input path. Filetypes returned can be from
+    `csv, excel, parquet, json, sdf`.
+    """
+    for name, extensions in EXTENSIONS_DICT.items():
+        for ext in extensions:
+            if path.endswith(ext):
+                return name
+
+
+# I was hoping to use something along the lines of this for checking kwargs
+# I dont think I'm correctly implementing it right now
+def _check_file_kwargs(
+    func: callable,
+    **kwargs: Any,
+) -> None:
+    """Check that the keyword arguments passed to the file reader are valid.
+
+    args:
+        func: function to check.
+        **kwargs: keyword arguments to check.
+    """
+    if not kwargs:
+        return
+
+    sig = inspect.signature(func)
+    allowed_kwargs = [p.name for p in sig.parameters.values() if p.kind == p.KEYWORD_ONLY]
+    for kwarg in kwargs:
+        if kwarg not in allowed_kwargs:
+            raise TypeError(f"Invalid keyword argument '{kwarg}'")
+
+
+def open_df(path: str, **kwargs: Any) -> pd.DataFrame:
+    """Open a dataframe file whatever its filetype from
+    `csv, excel, parquet, json, sdf`.
+
+    args:
+        path: path to the file.
+        kwargs: keyword arguments to pass to the underlying reader.
```

Comment:
The docstring should `**kwargs` and not `kwargs` (the function arg remains the same as it is).

It's important so the doc can be correctly built.